pkg/image/file_catalog.go

```diff
@@ -107,3 +209,27 @@ func (c *FileCatalog) FileContents(f file.Reference) (io.ReadCloser, error) {
 
 	return catalogEntry.Contents(), nil
 }
+
+func fileExtensions(p string) []string {
+	var exts []string
+	p = strings.TrimSpace(p)
+
+	// ignore oddities
+	if strings.HasSuffix(p, ".") {
+		return exts
+	}
+
+	// ignore directories
+	if strings.HasSuffix(p, "/") {
+		return exts
+	}
+
+	// ignore . which indicate a hidden file
+	p = strings.TrimLeft(path.Base(p), ".")
```

Comment:
The comment says we are ignoring hidden files here, but the later logic still looks like that when `.` is found we append the extension of the hidden file.

Reply:
the later logic starts from the end of the string and moves to the front of the string, reporting all extensions. So for instance, `file.tar.gz` would result in `.tar.gz` and `.gz` being found in the below loop. This "trim" protection is protecting against input like `.file.tar.gz` or `...file.tar.gz` so that we still only return `.tar.gz` + `gz` (and not something like `.file.tar.gz`)